ports/zephyr/src/usbd.c

```diff
@@ -0,0 +1,167 @@
+/*
+* This file is part of the MicroPython project, http://micropython.org/
+*
+* The MIT License (MIT)
+*
+* Copyright (c) 2024-2025 MASSDRIVER EI (massdriver.space)
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*/
+#include <stdint.h>
+
+#include <zephyr/device.h>
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/bos.h>
+
+#include "../mpconfigport.h"
```

Comment:
This file should not be included directly.  Instead include `py/mpconfig.h`.

Reply:
That doesnt work because it's a private

Reply:
I'm not sure what you mean by "private src"?

Note that `mpconfigport.h` may not be the correct file, sometimes it could be `mpconfigport_minimal.h`, or another custom file.  We need to find a way to make the include of `py/mpconfig.h` work.

Reply:
Yes, I understand the issue, but I am not sure how to fix it,
It is in this section:
```
target_sources(app PRIVATE
    src/zephyr_start.c
    src/zephyr_getchar.c
    src/usbd.c
)
```
so it doesnt have access to the `py/mpconfig.h` file in its include path (not exactly sure why, but it doesnt). How do you suggest solving this? moving it to zephyr_usbd.c one folder up?

Reply:
It feels like this is a layering issue:
1. zephyr itself
2. these files in `ports/zephyr/src` which are a slight extension to zephyr, to add a little more functionality for use by MicroPython
3. the MicroPython port to zephyr, sitting on top of the above

So, layer 2 should not know anything about layer 3.  If there's anything in layer 2 that needs layer 3, it should either be abstracted out somehow (eg a config option passed from 3 into 2) or moved to layer 3.

Why exactly is this config file needed?  Maybe it was for the VID/PID which were originally in that file, but now moved to Kconfig?

Reply:
You are right, it is not even necessary anymore...

Reply:
Done